src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -19,674 +19,557 @@
+        if (!pluginName.matches("[a-zA-Z0-9_\\-]+")) {
+            throw new ElasticsearchIllegalArgumentException("Plugin's name is invalid");
         }
 
-        static PluginHandle parse(String name) {
-            String[] elements = name.split("/");
-            // We first consider the simplest form: pluginname
-            String repo = elements[0];
-            String user = null;
-            String version = null;
-
-            // We consider the form: username/pluginname
-            if (elements.length > 1) {
-                user = elements[0];
-                repo = elements[1];
-
-                // We consider the form: username/pluginname/version
-                if (elements.length > 2) {
-                    version = elements[2];
-                }
-            }
-
-            if (repo.startsWith("elasticsearch-")) {
-                // remove elasticsearch- prefix
-                String endname = repo.substring("elasticsearch-".length());
-                return new PluginHandle(endname, version, user, repo);
+        for (String name : LocalPluginRepository.FORBIDDEN_NAMES) {
+            if (name.equalsIgnoreCase(pluginName)) {
+                throw new ElasticsearchIllegalArgumentException("Plugin's name cannot be equal to '" + pluginName + "'");
             }
+        }
+    }
 
-            if (name.startsWith("es-")) {
-                // remove es- prefix
-                String endname = repo.substring("es-".length());
-                return new PluginHandle(endname, version, user, repo);
-            }
+    /**
+     * Prints "no version" when the version is null or empty.
+     *
+     * @param version the plugin's version
+     * @return plugin's version or "no version"
+     */
+    private static String checkVersion(String version) {
+        return Strings.hasText(version) ? version : "no version";
+    }
 
-            return new PluginHandle(repo, version, user, repo);
-        }
+    protected static PluginRepository installedPluginsRepository(Environment env, Settings settings) {
+        return localRepository("local").environment(env).settings(settings).build();
     }
 
+    protected static PluginRepository defaultPluginRepository(Path downloadDir, TimeValue timeout, PluginDownloader.Listener listener, String userAgent, String esVersion) {
+        return chain("default")
+                .addRepository(
+                        urlRepository("elasticsearch")
+                                .url("http://download.elasticsearch.org/[organisation]/[name]/[name]-[version].zip")
+                                .url("http://download.elasticsearch.org/[organisation]/elasticsearch-[name]/elasticsearch-[name]-[version].zip")
+                                .url("http://download.elasticsearch.org/elasticsearch/[name]/[name]-latest.zip")
+                                .downloadDir(downloadDir)
+                                .listener(listener)
+                                .timeout(timeout)
```

Comment:
I wonder if `downloadDir(downloadDir)`,  `listener(listener)` and `timeout(timeout)` should be common to all repositories? I don't have a strong opinion about this but ideally, I think we should be able to define in `ChainedPluginRepository` a `DEFAULT` static variable which is built with:

```java
static final public PluginRepository DEFAULT = chain("default")
    .addRepository(
        urlRepository("elasticsearch")
            .url("http://download.elasticsearch.org/[organisation]/[name]/[name]-[version].zip")
            .url("http://download.elasticsearch.org/[organisation]/elasticsearch-[name]/elasticsearch-[name]-[version].zip")
            .url("http://download.elasticsearch.org/elasticsearch/[name]/[name]-latest.zip")
            .userAgent(PluginManager.class.getSimpleName())
            .elasticsearchVersion(Version.CURRENT.toString())
            .build()
    )
    .addRepository(
        urlRepository("central")
            .url("https://repo1.maven.org/maven2/[organisation]/[name]/[version]/[name]-[version].zip")
            .url("https://repo1.maven.org/maven2/org/elasticsearch/elasticsearch-[name]/[version]/elasticsearch-[name]-[version].zip")
            .url("http://search.maven.org/remotecontent?filepath=[organisation]/[name]/[version]/[name]-[version].zip")
            .url("http://search.maven.org/remotecontent?filepath=org/elasticsearch/elasticsearch-[name]/[version]/elasticsearch-[name]-[version].zip")
            .build()
)
    .addRepository(
        urlRepository("sonatype")
            .url("https://oss.sonatype.org/service/local/repositories/releases/content/[organisation]/[name]/[version]/[name]-[version].zip")
            .url("https://oss.sonatype.org/service/local/repositories/releases/content/[organisation]/elasticsearch-[name]/[version]/elasticsearch-[name]-[version].zip")
            .build()
    )
    .addRepository(
        urlRepository("github")
            .url("https://github.com/[organisation]/[name]/archive/[version].zip")
            .url("https://github.com/[organisation]/[name]/archive/master.zip")
            .url("https://github.com/[organisation]/elasticsearch-[name]/archive/[version].zip")
            .url("https://github.com/[organisation]/elasticsearch-[name]/archive/master.zip")
            .build()
    )
    .build();
```